tests/bsim/bluetooth/audio/src/tbs_test.c

```diff
@@ -328,22 +329,85 @@ static void test_tbs_server_only(void)
 	test_set_status_flags();
 }
 
-static void test_main(void)
+static void init(void)
 {
+	const struct bt_tbs_register_param gtbs_param = {
+		.provider_name = "Generic TBS",
+		.uci = "un000",
+		.uri_schemes_supported = "tel,skype",
+		.gtbs = true,
+		.authorization_required = false,
+		.technology = BT_TBS_TECHNOLOGY_3G,
+		.supported_features = CONFIG_BT_TBS_SUPPORTED_FEATURES,
+	};
 	int err;
 
 	err = bt_enable(NULL);
 	if (err != 0) {
-		printk("Bluetooth init failed (err %d)\n", err);
+		FAIL("Bluetooth enable failed (err %d)\n", err);
+
 		return;
 	}
 
-	printk("Audio Client: Bluetooth initialized\n");
+	printk("Bluetooth initialized\n");
 
 	bt_conn_cb_register(&conn_callbacks);
+	if (err != 0) {
+		FAIL("Failed to register conn CBs (err %d)\n", err);
+
+		return;
+	}
+
 	bt_le_scan_cb_register(&common_scan_cb);
+	if (err != 0) {
+		FAIL("Failed to register scan CBs (err %d)\n", err);
+
+		return;
+	}
+
 	bt_tbs_register_cb(&tbs_cbs);
```

Comment:
```suggestion
	err = bt_tbs_register_cb(&tbs_cbs);
	if (err != 0) {
		FAIL("Failed to register tbs CBs (err %d)\n", err);

		return;
	}
```

Reply:
This is has `void` as the return value still :)

Reply:
Sorry, my bad. I just grouped it with the others in my head :D

Reply:
Yeah, I don't blame you. It probably should have a return value added though